tests/test_client.py

```diff
@@ -188,6 +188,64 @@ def test_call_stub(self, MockSession, MockAdapter, mock_build_request_object, mo
         assert expected_response == actual_response
 
 
+    @mock.patch('apiron.client.Timeout')
+    @mock.patch('apiron.client.ServiceCaller.get_adapted_session')
+    @mock.patch('apiron.client.ServiceCaller.build_request_object')
+    @mock.patch('requests.adapters.HTTPAdapter', autospec=True)
+    @mock.patch('requests.Session', autospec=True)
+    def test_call_stub_dynamic(self, MockSession, MockAdapter, mock_build_request_object, mock_get_adapted_session, mock_timeout):
+        """
+        Test getting a response for a ``StubEndpoint`` using a dynamic response
+        """
+        service = mock.Mock()
+        service.get_hosts.return_value = ['http://host1.biz']
+
+        stub_endpoint = mock.Mock()
+        def stub_response(**kwargs):
+            return 'stub response'
+        stub_endpoint.stub_response = stub_response
+
+        actual_response = ServiceCaller.call(service, stub_endpoint)
+
+        expected_response = 'stub response'
+
+        assert expected_response == actual_response
```

Comment:
I'm tempted to pull the stub tests into their own class, like `TestClientStub` or something. Rationale: (1) there are a few of them now, and (2) they all use the same mocks, so we can just decorate the class.

Thoughts?

Reply:
Cool beans. I'll make the adjustment.